app/src/main/java/ch/epfl/sdp/appart/UserProfileActivity.java

```diff
@@ -282,4 +296,26 @@ private void setPictureToImageComponent() {
         database.accept(new GlideImageViewLoader(this, imageView,
                 this.sessionUser.getProfileImagePathAndName()));
     }
+
+    private void saveUserLocally() {
+        CompletableFuture<Bitmap> profileImgResult = new CompletableFuture<>();
+        CompletableFuture<Void> userRes = mViewModel.getCurrentUser();
+        userRes.exceptionally(e -> {
+            Toast.makeText(this, R.string.localSaveFailed_User, Toast.LENGTH_SHORT).show();
+            return null;
+        });
+        userRes.thenAccept(res -> {
+            database.accept(new GlideBitmapLoader(this, profileImgResult,
+                    mViewModel.getUser().getValue().getProfileImagePathAndName()));
+            profileImgResult
+                    .exceptionally(e -> {
+                        Toast.makeText(this, R.string.localSaveFailed_User,
+                                Toast.LENGTH_SHORT).show();
+                        return null;
+                    })
+                    .thenAccept(img -> {
+                        localdb.setCurrentUser(mViewModel.getUser().getValue(), img);
+                    });
+        });
+    }
```

Comment:
This looks a lot like savedLoggedUser in LoginActivity. Maybe this functionality could be extracted into a method.